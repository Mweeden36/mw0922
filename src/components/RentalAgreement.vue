<template>
  <div class="rental-agreement panel">
    <h1 class="rental-agreement__header">Rental Agreement</h1>
    <RentalAgreementItem
      v-for="(val, key) in rentalAgreement"
      :key="key"
      :itemType="key"
      :value="val"
    />
    <button @click="$emit('back')">Back to Rental Form</button>
  </div>
</template>

<script>
import RentalAgreementItem from "@/components/RentalAgreementItem.vue";

export default {
  name: "RentalAgreement",

  components: {
    RentalAgreementItem,
  },

  props: {
    rentalAgreement: {
      type: Object,
      required: true,
    },
  },

  methods: {
    formatCurrency(val) {
      return `$${val.toFixed(2)}`;
    },
  },
};
</script>

<style lang="scss" scoped>
.rental-agreement {
  width: 500px;
  .rental-agreement__header {
    padding-bottom: 10px;
  }
}
</style>
