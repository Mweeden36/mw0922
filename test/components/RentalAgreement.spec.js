import RentalAgreement from "../../src/components/RentalAgreement.vue";
import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

describe("RentalAgreement.vue", () => {
  const testAgreement = {
    toolCode: "JAKD",
    checkoutDate: "09/01/22",
    returnDate: "09/04/22",
    discountPercent: 4,
    chargeableDays: 3,
    dailyCharge: 1.99,
    prediscountAmount: 5.97,
    discountAmount: 0.24,
    finalAmount: 5.73,
  };

  it("should initialize with sensible defaults", () => {
    const { vm } = mount(RentalAgreement, {
      props: {
        rentalAgreement: testAgreement,
      },
    });
    expect(vm.rentalAgreement).toEqual(testAgreement);
  });

  it("should generate the correct html", () => {
    const wrapper = mount(RentalAgreement, {
      props: {
        rentalAgreement: testAgreement,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
