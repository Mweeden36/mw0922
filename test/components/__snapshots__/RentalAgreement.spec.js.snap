// Vitest Snapshot v1

exports[`RentalAgreement.vue > should generate the correct html 1`] = `
"<div class=\\"rental-agreement panel\\" data-v-5b29b027=\\"\\">
  <h1 class=\\"rental-agreement__header\\" data-v-5b29b027=\\"\\">Rental Agreement</h1>
  <div class=\\"rental-agreement-item toolCode\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Tool</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">DeWalt jackhammer</span></div>
  <div class=\\"rental-agreement-item checkoutDate\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Checkout Date</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">09/01/22</span></div>
  <div class=\\"rental-agreement-item returnDate\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Return Date</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">09/04/22</span></div>
  <div class=\\"rental-agreement-item discountPercent\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Discount Percent</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">4%</span></div>
  <div class=\\"rental-agreement-item chargeableDays\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Chargeable Days</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">3</span></div>
  <div class=\\"rental-agreement-item dailyCharge\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Daily Charge</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">$1.99</span></div>
  <div class=\\"rental-agreement-item prediscountAmount\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Pre-discount Amount</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">$5.97</span></div>
  <div class=\\"rental-agreement-item discountAmount\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Discount Amount</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">$0.24</span></div>
  <div class=\\"rental-agreement-item finalAmount\\" data-v-4db1ffff=\\"\\" data-v-5b29b027=\\"\\"><b class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">Final Amount</b><span class=\\"rental-agreement__label\\" data-v-4db1ffff=\\"\\">$5.73</span></div><button class=\\"button back\\" data-v-5b29b027=\\"\\"> Back to Rental Form </button>
</div>"
`;
